kube-runtime/src/events.rs

```diff
@@ -171,12 +172,18 @@ impl Recorder {
     /// Create a new recorder that can publish events for one specific object
     ///
     /// This is intended to be createad at the start of your controller's reconcile fn.
+    ///
+    /// Cluster scoped object reference defaults to "default" namespace.
     #[must_use]
-    pub fn new(client: Client, reporter: Reporter, reference: ObjectReference) -> Self {
-        let events = match reference.namespace.as_ref() {
-            None => Api::all(client),
-            Some(namespace) => Api::namespaced(client, namespace),
+    pub fn new(client: Client, reporter: Reporter, reference: ObjectReference, namespace_override: Option<&str>) -> Self {
+        let namespace = match namespace_override {
+            Some(namespace) => namespace,
+            None => match reference.namespace.as_ref() {
+                Some(namespace) => namespace,
+                None => "default",
+            }
```

Comment:
```suggestion
    pub fn new(client: Client, reporter: Reporter, reference: ObjectReference) -> Self {
        let events = Api::namespaced(client, reference.namespace.as_ref().unwrap_or_else(|| "default");
```

if that works, might need some String::from magic